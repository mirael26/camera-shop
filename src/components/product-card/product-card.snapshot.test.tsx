import { renderWithReduxAndRouter } from '../../test/helpers/render-with-redux-and-router';
import { productMock } from '../../test/mocks';
import ProductCard from './product-card';

test('ProductCard renders correctly', () => {
  const { asFragment } = renderWithReduxAndRouter(<ProductCard product={productMock}/>);
  expect(asFragment()).toMatchSnapshot();
});
