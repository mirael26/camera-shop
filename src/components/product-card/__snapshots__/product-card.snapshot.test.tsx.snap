// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<DocumentFragment>
  <div
    class="product-card"
    data-testid="product-card"
  >
    <div
      class="product-card__img"
    >
      <picture>
        <source
          srcset="img/content/orlenok.webp, img/content/orlenok@2x.webp 2x"
          type="image/webp"
        />
        <img
          alt="Орлёнок"
          height="240"
          src="img/content/orlenok.jpg"
          srcset="img/content/orlenok@2x.jpg 2x"
          width="280"
        />
      </picture>
    </div>
    <div
      class="product-card__info"
    >
      <div
        class="rate product-card__rate"
      >
        <ratingstars
          rating="3"
        />
        <p
          class="visually-hidden"
          data-testid="rating"
        >
          Рейтинг: 3
        </p>
        <p
          class="rate__count"
          data-testid="rate-count"
        >
          <span
            class="visually-hidden"
          >
            Всего оценок:
          </span>
          17
        </p>
      </div>
      <p
        class="product-card__title"
        data-testid="name"
      >
        Орлёнок
      </p>
      <p
        class="product-card__price"
        data-testid="price"
      >
        <span
          class="visually-hidden"
        >
          Цена:
        </span>
        19 970 ₽
      </p>
    </div>
    <div
      class="product-card__buttons"
    >
      <button
        class="btn btn--purple product-card__btn"
        type="button"
      >
        Купить
      </button>
      <a
        class="btn btn--transparent"
        href="/catalog/product/4"
      >
        Подробнее
      </a>
    </div>
  </div>
</DocumentFragment>
`;
